plugin/manager.go

```diff
@@ -0,0 +1,124 @@
+package plugin
+
+import (
+	"fmt"
+	"github.com/cloudquery/cloudquery/logging"
+	"github.com/hashicorp/go-plugin"
+	"github.com/rs/zerolog/log"
+	"os"
+	"os/exec"
+	"sync"
+)
+
+var (
+	doOnce   sync.Once
+	instance *Manager
+)
+
+type Manager struct {
+	activeClients map[string]*plugin.Client
+	registry      map[string]CQProvider
+
+	lock sync.RWMutex
```

Comment:
No need for the locks in the manager as we are not calling it from multiple go routines.